app/components/Form/ProjectEmissionIntensityReportForm.tsx

```diff
@@ -197,7 +238,9 @@ const ProjectEmissionsIntensityReport: React.FC<Props> = (props) => {
             formContext={{
               form: emissionIntensityReportFormChange?.newFormData,
             }}
-            uiSchema={emissionIntensityReportUiSchema}
+            uiSchema={createEmissionIntensityReportUiSchema(
+              emissionIntensityReportFormChange?.newFormData
```

Comment:
If the newFormData is `undefined` here, this will break the destructuring in the function definition. No need to pass all the newFormData, the `createEmissionIntensityReportUiSchema` should only have as parameters what it needs, just the two `emissionFunctionalUnit` and `productionFunctionalUnit`

```suggestion
              emissionIntensityReportFormChange?.newFormData?.emissionFunctionalUnit,
              emissionIntensityReportFormChange?.newFormData?.productionFunctionalUnit
```

Reply:
Absolutely right.